config/channels/in-memory-channel/deployments/dispatcher.yaml

```diff
@@ -24,6 +24,11 @@ metadata:
     app.kubernetes.io/version: devel
     app.kubernetes.io/part-of: knative-eventing
 spec:
+  strategy:
+    rollingUpdate:
+      maxSurge: 50%
+      maxUnavailable: 75%
+    type: RollingUpdate
```

Comment:
Just to making sure I understand what problem this solves.
If we for example scale from `2` to `6`, Kubernetes sees `4` unavailable Pods and immeditely marks the Deployment as [Available: false](https://pkg.go.dev/k8s.io/api/apps/v1?utm_source=godoc#DeploymentConditionType)?

Reply:
Scaling from `2` to `6` with these numbers does not trigger `MinimumReplicasUnavailable` dispatcher deployment status, but for example `1` to `10` does. With default values (`25%`/`25%`) scaling from `1` to `2` caused a short channels downtime.

Reply:
Is there anything we need to be aware of from the perspective of the `knative/operator`. Once Eventing is installed via the operator.. that guy owns the entire deployment. 

Especially when setting "HA" to just `replicas: 1`:
https://github.com/knative/operator/blob/main/pkg/apis/operator/v1alpha1/common.go#L150


Should we expose better knobs for this, for the operator?

Reply:
/cc @houshengbo for Operator WG perspective

Reply:
@tzununbekov any comments ?

Reply:
@matzew totally forgot about the operator! I see these numbers as a subject for fine-tuning in each particular setup environment. While default values are ok in the cluster where no active dispatcher scaling is expected, in the prod cluster with a load they may cause issues. New numbers in this PR were empirically found but may also need to be tuned. Given all this, I'd say that having knobs for update strategy in operator may be a good idea.

Reply:
@tzununbekov can you follow up on that on the `knative/operator` repo ?